File: app/src/androidTest/java/com/glsebastiany/bakingapp/core/injection/MockedApplicationModule.java
package com.glsebastiany.bakingapp.core.injection;

import com.glsebastiany.bakingapp.MyApplication;
import com.glsebastiany.bakingapp.core.repository.DecoratedRecipesRepository;
import com.glsebastiany.bakingapp.injection.module.ApplicationModule;
import com.glsebastiany.bakingapp.injection.scopes.ApplicationContext;
import com.glsebastiany.bakingapp.repository.RecipesRepository;

import dagger.Module;
import dagger.Provides;

@Module
public class MockedApplicationModule extends ApplicationModule {

     private MyApplication application;

     public MockedApplicationModule(MyApplication application){
         super(application);
         this.application = application;
        }

    @Provides
    @ApplicationContext
    public RecipesRepository provideRecipeRepository() {
        return new DecoratedRecipesRepository(application);
    }

}

File: app/src/androidTest/java/com/glsebastiany/bakingapp/core/injection/MockedApplicationModule.kt
package com.glsebastiany.bakingapp.core.injection

import com.glsebastiany.bakingapp.MyApplication
import com.glsebastiany.bakingapp.core.repository.DecoratedRecipesRepository
import com.glsebastiany.bakingapp.injection.module.ApplicationModule
import com.glsebastiany.bakingapp.injection.scopes.ApplicationContext
import com.glsebastiany.bakingapp.repository.RecipesRepository
import dagger.Module
import dagger.Provides

@Module
class MockedApplicationModule(private val application: MyApplication) : ApplicationModule(application) {

    @Provides
    @ApplicationContext
    override fun provideRecipeRepository(): RecipesRepository = DecoratedRecipesRepository(application)

}